java/beakerx support

scottdraves opened the pull request:
this works: https://github.com/twosigma/beakerx/blob/master/doc/groovy/3D.ipynb

maartenbreddels commented:
Great, I also see you had to get some parts of ipywebrtc in.

maartenbreddels commented:
This seems to work nicely with Clojure:
```clojure
%classpath config resolver jitpack.io https://jitpack.io
%classpath add mvn com.github.twosigma:ipyvolume:lmitusinski~beakerx-widget-SNAPSHOT
(import ipyvolume.PyLab)

(defn ball 
    [size radius]
    (let [data           (make-array Float/TYPE size size size)
          square         #(* % %)
          center         #(- % (/ size 2))
          sum            (partial reduce +)
          sum-of-squares #(sum (map square %))]
        (doseq [i (range size)
                j (range size)
                k (range size)
                :let [[x y z] (map center [i j k])] 
                :when (> (square radius) 
                         (sum-of-squares [x y z]))]
             (aset data i j k (float 1)))
        data))
(PyLab/volShow (ball 32 12))
```

scottdraves commented:
as soon as we get pythreejs working, will look into the test failure:
```
_____________________________ test_threejs_version _____________________________
    def test_threejs_version():
        # a quick check, as a reminder to change if threejs version is updated
        configpath = os.path.join(os.path.abspath(ipyvolume.__path__[0]), "..", "js", "package.json")
        with open(configpath) as f:
            config = json.load(f)
        major, minor = ipyvolume._version.__version_threejs__.split(".")
        major_js, minor_js, patch_js = config['dependencies']['three'][1:].split(".")
        version_msg = "version in python and js side for three js conflect: %s vs %s" % (
            ipyvolume._version.__version_threejs__, config['dependencies']['three'])
>       assert (major == major_js) and (minor == minor_js), version_msg
E       AssertionError: version in python and js side for three js conflect: 0.85 vs ^0.91.0
E       assert ('0' == '0'
E           0 and '85' == '91'
E         - 85
E         + 91)
ipyvolume/test_all.py:366: AssertionError
```

maartenbreddels commented:
I just fixed that, if you rebase against master it should work.
Btw, how is distribution and installation managed, did you think about it? Apart from a conda package, what is the way this should be distributed?

scottdraves commented:
thanks, it's green now.

scottdraves commented:
for distribution, jitpack does fine for the java part afaik.

mprogram commented:
I've got compiled `ipyvolume-java-1.0-SNAPSHOT.jar` in the `ipyvolume-java` folder. Any suggestion how do one needs to integrate it further into the Debian system I'm running to look at it? I've passed the opening example to `myBinder`, unfortunately, with no visuals.

mprogram commented:
Yes, I did. I'm off my testing environment at the moment due to real life demands, however, I'm more than happy to provide a universal link to query _virtually any notebook_ hosted on GitHub straight into myBinder environment (_watching the 'Build logs' by clicking the 'show' link at the top would be a good thing to do as it hangs quite often_): 
https://mybinder.org/v2/gh/twosigma/beakerx/master?filepath=doc/groovy/3D.ipynb
